pkg/storage/unified/sql/backend.go

```diff
@@ -631,6 +632,79 @@ func (b *backend) listLatest(ctx context.Context, req *resourcepb.ListRequest, c
 	return iter.listRV, err
 }
 
+// ListModifiedSince will return all resources that have changed since the given resource version.
+// If a resource has changes, only the latest change will be returned.
+func (b *backend) ListModifiedSince(ctx context.Context, key resource.NamespacedResource, sinceRv int64) (int64, iter.Seq2[*resource.ModifiedResource, error]) {
+	var latestRv int64
+	err := b.db.WithTx(ctx, ReadCommittedRO, func(ctx context.Context, tx db.Tx) error {
+		var err error
+		latestRv, err = b.fetchLatestRV(ctx, tx, b.dialect, key.Group, key.Resource)
+		return err
+	})
+	if err != nil {
+		return 0, func(yield func(*resource.ModifiedResource, error) bool) {
+			yield(nil, err)
+		}
+	}
+
+	// track seen resources to avoid duplicates
+	seen := make(map[string]bool)
+
+	seq := func(yield func(*resource.ModifiedResource, error) bool) {
+		err := b.db.WithTx(ctx, RepeatableRead, func(ctx context.Context, tx db.Tx) error {
+			var err error
+
+			query := sqlResourceListModifiedSinceRequest{
+				SQLTemplate: sqltemplate.New(b.dialect),
+				Namespace:   key.Namespace,
+				Group:       key.Group,
+				Resource:    key.Resource,
+				SinceRv:     sinceRv,
+			}
+
+			rows, err := dbutil.QueryRows(ctx, tx, sqlResourceHistoryListModifiedSince, query)
+			if rows != nil {
+				defer func() {
+					if cerr := rows.Close(); cerr != nil {
+						b.log.Warn("listSinceModified error closing rows", "error", cerr)
+					}
+				}()
+			}
+			if err != nil {
+				yield(nil, err)
+				return nil
+			}
```

Comment:
If `err` is set, rows will always be nil, checking error first.
```suggestion
			if err != nil {
				yield(nil, err)
				return nil
			}
			if rows != nil {
				defer func() {
					if cerr := rows.Close(); cerr != nil {
						b.log.Warn("listSinceModified error closing rows", "error", cerr)
					}
				}()
			}
```